src/openms/include/OpenMS/CHEMISTRY/MASSDECOMPOSITION/IMS/RealMassDecomposer.h

```diff
@@ -126,7 +126,7 @@ namespace OpenMS
         Decomposer to be used for exact decomposing using
         integer arithmetic.
       */
-      std::auto_ptr<integer_decomposer_type> decomposer_;
+      std::shared_ptr<integer_decomposer_type> decomposer_;
```

Comment:
well basically I was worried that `unique_ptr` may not work out of the box. See https://stackoverflow.com/questions/16030081/copy-constructor-for-a-class-with-unique-ptr#16030118

- Since the unique_ptr can not be shared, you need to either deep-copy its content or convert the unique_ptr to a shared_ptr.

so no real reason but I felt shared_ptr is less work and works as-is